powerauth-java-server/src/main/java/io/getlime/security/powerauth/app/server/service/behavior/tasks/TemporaryKeyBehavior.java

```diff
@@ -278,21 +279,23 @@ private TemporaryKeyResult obtainTemporaryKeyResult(TemporaryPublicKeyRequestCla
                 return result;
             } else {
 
-                final String appId = applicationVersionEntity.getApplication().getId();
+                final Long appId = applicationVersionEntity.getApplication().getRid();
```

Comment:
🤔 Is this change needed (the `id` should be also unique)? If so, I suggest renaming the variable, because it is confusing now (`id` vs `rid`).

Reply:
We can refactor it... I used `rid` here because the comparison could be slightly faster...